doc/build-helpers/images/dockertools.section.md

```diff
@@ -561,6 +561,8 @@ This allows the function to produce reproducible images.
   This allows scripts that perform installation in `/` to work as expected.
   This can be seen as an equivalent of `RUN ...` in a `Dockerfile`.
 
+This does not work under `darwin`!
```

Comment:
I am not sure about this, as I don't see a similar mention elsewhere in the documentation - not sure if that's simply because there's a convention of not mentioning incompatibilities due to bugs or implementation deficiencies, or simply because the situation doesn't occur that much. Regardless, I think this needs to be aligned with the paragraph above.

Reply:
@NixOS/documentation-team Do you guys have input on this? Is there a convention we're not aware of? And would you think this is fitting? Any input is appreciated! 🫶

Reply:
Thanks for the ping. I'm not excited about leaving documentation that may get out of date. Sure, reading that in the manual before running into the problem would be great, but I'd err towards maintainability here. If the error is self-explanatory, it should be good enough. I strongly recommend explaining in the error message *why* it doesn't work on Darwin and what to do about it (e.g. workarounds or upvoting an issue so we can keep track of interest).

Reply:
I've removed the documentation part and have added a bit more text to the assert error message. Thanks for the input!

Really helpful to get those kind of insights when only starting the journey of contributing. Thank you!